### PR TITLE
feat: add wallet contract

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

Add a Wallet system contract:
1. store encrypted transaction metadata
2. store username request information to help masternode voters distinguish 

## What was done?

Add Wallet system contract.  This is called "wallet" because it stores wallet-level 

Preliminary documentation on the contract:
**Transaction Metadata**: https://docs.google.com/document/d/1aLDZmum_MOWF_QVkFbkAVR-m3Wx-FqINiVhCRYBMZXw/edit?tab=t.0
1. This document still refers to using CBOR for storage before encryption, but it will be changed to ProtoBuf.

**Identity Verify** - This document stores a normalizedLabel and normalizedParentDomainName and a url to a social media post or some kind of post that helps prove the online identity of a platform identity.  It supports one index to allow matching an identity with a particular username (normalizedLabel, normalizedParentDomainName).

This documentation will need to be converted to a DIP. 

## How Has This Been Tested?

This was not tested.  My local systems would not run the `yarn setup` or `yarn start` successfully to execute the tests, perhaps due to configuration issues.

## Breaking Changes

There should be no breaking changes.

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Release Notes

- **New Features**
  - Introduced the `wallet-contract` package for managing wallet data, including a JSON schema for documents.
  - Added ESLint and Mocha configurations for improved code quality and testing.
  - Implemented comprehensive unit tests for Wallet Contract functionalities.
  - Defined new schemas for `txMetadata` to enhance document structure.
  - Added constants and functions for error handling and version management.

- **Documentation**
  - Created a README.md file detailing installation, usage, and contribution guidelines.
  - Added MIT License to the package.

- **Bug Fixes**
  - Enhanced error handling for version mismatches and schema validation in the wallet contract.

### CODE REVIEW

File: config/local_seed.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (shumkov) [inline config/local_seed.json:1]: Why do we have these two files?

File: packages/wallet-contract/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@dashevo/wallet-contract",
+  "version": "1.4.1",
```
Comment (shumkov) [inline packages/wallet-contract/package.json:3]: 1.5.0

File: packages/wallet-contract/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "wallet-contract"
+description = "Wallet data contract schema and tools"
+version = "1.4.1"
```
Comment (shumkov) [inline packages/wallet-contract/Cargo.toml:4]: 1.5.0

File: packages/wallet-contract/schema/v1/wallet-contract-documents.json [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    },
+    "required": [
+      "keyIndex",
+      "encryptionKeyIndex",
+      "encryptedMetadata",
+      "$createdAt"
+    ],
+    "additionalProperties": false
+  },
+  "identityVerify": {
```
Comment (shumkov) [inline packages/wallet-contract/schema/v1/wallet-contract-documents.json:55]: What is it for? How is it related to the wallet?

File: packages/wallet-contract/test/bootstrap.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+const {
+  default: loadWasmDpp,
+} = require('@dashevo/wasm-dpp');
+
+use(dirtyChai);
+use(sinonChai);
+
+exports.mochaHooks = {
+  beforeAll: loadWasmDpp,
+
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/wallet-contract/test/bootstrap.js:16]: `loadWasmDpp` is async too

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by HashEngineering** [comment]
Thank you.  I started this PR when 1.4.1 the latest version, so some of these things are out of date. I will need to rebase against 1.6-dev and then make required changes.

File: config/local_seed.json (hunk 1 above)
Comment (HashEngineering) [reply config/local_seed.json:1]: oh, that is a mistake

File: packages/wallet-contract/schema/v1/wallet-contract-documents.json (hunk 4 above)
Comment (HashEngineering) [reply packages/wallet-contract/schema/v1/wallet-contract-documents.json:55]: In the DashPay app, when a user requests a contested username, we allow the user to specify a social media post to "prove" an identity.

If two people wanted `bob` each can specify a url to a social media post that says something like "I am the real bob and my identity id is `Abs19....47k`."

Masternode owners using DashPay can view this see this URL in the username request details page of the Username Voting screen. 

This contact might not be the best place for it.  If it isn't, I am not sure if it should be in a new contract or the dashpay contract.

File: config/local_seed.json (hunk 1 above)
Comment (HashEngineering) [reply config/local_seed.json]: fixed

File: packages/wallet-contract/package.json (hunk 2 above)
Comment (HashEngineering) [reply packages/wallet-contract/package.json:3]: reset

File: packages/wallet-contract/Cargo.toml (hunk 3 above)
Comment (HashEngineering) [reply packages/wallet-contract/Cargo.toml:4]: reset

File: packages/wallet-contract/schema/v1/wallet-contract-documents.json (hunk 4 above)
Comment (shumkov) [reply packages/wallet-contract/schema/v1/wallet-contract-documents.json:55]: Yeah, it sounds like something else. Let's remove this from this PR and think more about it?

**Pull request merged**